qa/suites/upgrade/quincy-p2p/quincy-p2p-parallel/point-to-point-upgrade.yaml

```diff
@@ -0,0 +1,205 @@
+overrides:
+  ceph:
+    log-ignorelist:
+    - reached quota
+    - scrub
+    - osd_map_max_advance
+    - wrongly marked
+    - FS_DEGRADED
+    - POOL_APP_NOT_ENABLED
+    - CACHE_POOL_NO_HIT_SET
+    - POOL_FULL
+    - SMALLER_PG
+    - pool\(s\) full
+    - OSD_DOWN
+    - missing hit_sets
+    - CACHE_POOL_NEAR_FULL
+    - PG_AVAILABILITY
+    - PG_DEGRADED
+    - application not enabled
+    - cache pools at or near target size
+    - filesystem is degraded
+    - OBJECT_MISPLACED
+    ### ref: https://tracker.ceph.com/issues/40251
+    #removed see ^ - failed to encode map
+
+    fs: xfs
+
+    conf:
+      global:
+        mon_warn_on_pool_no_app: false
+        mon_mds_skip_sanity: true
+      mon:
+        mon debug unsafe allow tier with nonempty snaps: true
+      osd:
+        osd map max advance: 1000
+        osd_class_default_list: "*"
+        osd_class_load_list: "*"
+      client:
+        rgw_crypt_require_ssl: false
+        rgw crypt s3 kms backend: testing
+        rgw crypt s3 kms encryption keys: testkey-1=YmluCmJvb3N0CmJvb3N0LWJ1aWxkCmNlcGguY29uZgo= testkey-2=aWIKTWFrZWZpbGUKbWFuCm91dApzcmMKVGVzdGluZwo=
+roles:
+- - mon.a
+  - mds.a
+  - osd.0
+  - osd.1
+  - osd.2
+  - mgr.x
+- - mon.b
+  - mon.c
+  - osd.3
+  - osd.4
+  - osd.5
+  - client.0
+- - client.1
+openstack:
+- volumes: # attached to each instance
+    count: 3
+    size: 30 # GB
+tasks:
+- print: "****  done quincy about to install v17.2.0 "
+- install:
+    tag: v17.2.0
+    # line below can be removed its from jewel test
+    #exclude_packages: ['ceph-mgr','libcephfs2','libcephfs-devel','libcephfs-dev', 'librgw2']
+- print: "**** done v17.2.0 install"
+- ceph:
+   fs: xfs
+   add_osds_to_crush: true
+- print: "**** done ceph xfs"
+- sequential:
+   - workload
+- print: "**** done workload v17.2.0"
+
+
+#######  upgrade to v17.2.3
+- install.upgrade:
+    #exclude_packages: ['ceph-mgr','libcephfs2','libcephfs-devel','libcephfs-dev']
+    mon.a:
+      tag: v17.2.3
+    mon.b:
+      tag: v17.2.3
+- parallel:
+   - workload_quincy_point
+   - upgrade-sequence_quincy
+- print: "**** done parallel quincy v17.2.3"
+
+####  upgrade to latest quincy
+- install.upgrade:
+    mon.a:
+    mon.b:
+- parallel:
+   - workload_quincy_latest
+   - upgrade-sequence_quincy
+- print: "**** done parallel quincy branch"
+
+#######################
+workload:
+   sequential:
+   - workunit:
+       clients:
+         client.0:
+           - suites/blogbench.sh
+
+workload_quincy_point:
+   full_sequential:
+   - workunit:
+       branch: quincy
+       tag: v17.2.3
+       clients:
+         client.1:
+         - rados/test.sh
+         - cls
+       env:
+         CLS_RBD_GTEST_FILTER: '*:-TestClsRbd.snapshots_namespaces'
+   - print: "**** done rados/test.sh &  cls workload_quincy_point"
+   - sequential:
+     - rgw: [client.0]
+     - print: "**** done rgw workload_quincy_point"
+     - s3tests:
+         client.0:
+           force-branch: ceph-quincy
```

Comment:
My change might not have worked since we're forcing the branch here to `ceph-quincy`...

Reply:
yeah try `ceph-quincy-17.2.3` there

Reply:
Alright, trying it again here with `ceph-quincy-17.2.3`:
http://pulpito.front.sepia.ceph.com/lflores-2022-09-20_22:40:13-upgrade:quincy-p2p:quincy-p2p-parallel-quincy-distro-default-smithi/7039486/

Reply:
@cbodley bit different of a failure this time:

```
2022-09-21T00:10:45.458 INFO:teuthology.orchestra.run.smithi055.stderr:======================================================================
2022-09-21T00:10:45.458 INFO:teuthology.orchestra.run.smithi055.stderr:ERROR: s3tests_boto3.functional.test_s3.test_bucket_policy_put_obj_s3_noenc
2022-09-21T00:10:45.459 INFO:teuthology.orchestra.run.smithi055.stderr:----------------------------------------------------------------------
2022-09-21T00:10:45.459 INFO:teuthology.orchestra.run.smithi055.stderr:Traceback (most recent call last):
2022-09-21T00:10:45.459 INFO:teuthology.orchestra.run.smithi055.stderr:  File "/home/ubuntu/cephtest/s3-tests/virtualenv/lib/python3.6/site-packages/nose/case.py", line 198, in runTest
2022-09-21T00:10:45.460 INFO:teuthology.orchestra.run.smithi055.stderr:    self.test(*self.arg)
2022-09-21T00:10:45.460 INFO:teuthology.orchestra.run.smithi055.stderr:  File "/home/ubuntu/cephtest/s3-tests/s3tests_boto3/functional/test_s3.py", line 12980, in test_bucket_policy_put_obj_s3_noenc
2022-09-21T00:10:45.461 INFO:teuthology.orchestra.run.smithi055.stderr:    response = client.put_object(Bucket=bucket_name, Key=key1_str, ServerSideEncryption='AES256')
2022-09-21T00:10:45.461 INFO:teuthology.orchestra.run.smithi055.stderr:  File "/home/ubuntu/cephtest/s3-tests/virtualenv/lib/python3.6/site-packages/botocore/client.py", line 508, in _api_call
2022-09-21T00:10:45.461 INFO:teuthology.orchestra.run.smithi055.stderr:    return self._make_api_call(operation_name, kwargs)
2022-09-21T00:10:45.462 INFO:teuthology.orchestra.run.smithi055.stderr:  File "/home/ubuntu/cephtest/s3-tests/virtualenv/lib/python3.6/site-packages/botocore/client.py", line 911, in _make_api_call
2022-09-21T00:10:45.462 INFO:teuthology.orchestra.run.smithi055.stderr:    raise error_class(parsed_response, operation_name)
2022-09-21T00:10:45.462 INFO:teuthology.orchestra.run.smithi055.stderr:botocore.exceptions.ClientError: An error occurred (InvalidArgument) when calling the PutObject operation: Unknown
```

Reply:
It's still not using the right filter:
`!fails_on_rgw,!lifecycle_expiration,!fails_strict_rfc2616,!test_of_sts,!webidentity_test,!fails_with_subdomain`

Reply:
i see from the log that the `ceph-quincy-17.2.3` tests had passed before this. are you sure the radosgws finished upgrading before it ran these `ceph-quincy` tests?

Reply:
@cbodley interesting.. let me verify

Reply:
@cbodley you are correct! The s3tests passed now in 17.2.3:
`2022-09-20T23:40:34.234 INFO:teuthology.task.print:**** done s3tests workload_quincy_point`

But now they are failing with quincy_latest. (This debug message prints right before the rgw workload runs):
`*** done rgw workload_quincy_latest`

Reply:
At least we're getting somewhere

Reply:
@cbodley do you have any idea why s3tests are now failing in quincy_latest?

It passed in the point version with the modified branch you created.

Reply:
was http://qa-proxy.ceph.com/teuthology/lflores-2022-09-20_22:40:13-upgrade:quincy-p2p:quincy-p2p-parallel-quincy-distro-default-smithi/7039486/teuthology.log your latest result here?

the failing s3tests ran with this filter:

> virtualenv/bin/python -m nose -w /home/ubuntu/cephtest/s3-tests -v -a '!fails_on_rgw,!lifecycle_expiration,!fails_strict_rfc2616,!test_of_sts,!webidentity_test,!fails_with_subdomain'

this relates to my earlier comment:

> other successful quincy runs are using this s3tests filter:
>
> > !fails_on_rgw,!lifecycle_expiration,!fails_strict_rfc2616,!test_of_sts,!webidentity_test,!fails_with_subdomain,!sse-s3
>
>but your job used the filter:
>
> > !fails_on_rgw,!lifecycle_expiration,!fails_strict_rfc2616,!test_of_sts,!webidentity_test,!fails_with_subdomain
>
> looking at https://github.com/ceph/ceph/blob/quincy/qa/tasks/s3tests.py#L433-L441, it's not clear to me why the !sse-s3 tag isn't being added here

qa/tasks/s3tests.py didn't start adding that `!sse-s3` filter until sse-s3 the support was added. so this must be running the old 17.2.3 suite-branch

Reply:
@cbodley this is the branch that quincy_latest is using to run the s3test workload: `ceph-quincy`. So to my understanding, if it isn't using the correct filter, perhaps `ceph-quincy` is not up to date?

Reply:
the ceph-quincy and ceph-quincy-17.2.3 branches live in the https://github.com/ceph/s3-tests repo, but the filter itself comes from our suite branch in https://github.com/ceph/ceph/blob/quincy/qa/tasks/s3tests.py#L433-L441. that part doesn't exist on the 17.2.3 suite branch: https://github.com/ceph/ceph/blob/v17.2.3/qa/tasks/s3tests.py#L432-L434